docs/installation/prepare/microcontrollers/_index.md

```diff
@@ -1,387 +1,353 @@
+3. [Install the Micro-RDK](#install-the-micro-rdk) and [create](#create-a-new-robot) and [connect to](#connect-to-your-esp32) a new robot
+
+### Hardware Requirements
+
+You need an an Espressif ESP32 microcontroller to use the micro-RDK.
+Viam recommends purchasing the ESP32 with a development board: see [development kit options](https://www.espressif.com/en/products/devkits).
+
+* **Minimal configuration:** 384kB Ram 4MB flash
+* **Recommended configuration:** 384kB Ram + 8MB SPIRAM + 4MB Flash
+
+### Software Requirements
+
+The micro-RDK is written in Rust.
+To be able to program the ESP32 on macOS and Linux systems, you must install the following software on your development machine:
+
+#### Install Dependencies
+
+{{< tabs >}}
+{{% tab name="Linux (Ubuntu)" %}}
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+sudo apt-get install git libssl-dev dfu-util libusb-1.0-0 libudev-dev
+```
+
+{{% /tab %}}
+{{% tab name="macOS" %}}
+
+If you haven't already, install Homebrew:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+Then, install `dfu-util`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+brew install dfu-util
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+#### Install `Rust`
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+See the [Rust Installation guide](https://www.rust-lang.org/tools/install) for more information and other installation methods.
+
+#### Install `cargo-generate` with `cargo`
+
+ `cargo` installs automatically when downloading Rust with rustup.
+
+Run the following command to install `cargo-generate`:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install cargo-generate
+```
+
+#### Install `espflash`
+
+Run the following command to install `espflash`
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+cargo install espflash
+```
+
+## Set up your development enviroment
+
+### Use Viam's Canon CLI utility
+
+[Canon](https://github.com/viamrobotics/canon) is a CLI utility for managing a Docker-based canonical environment.
+
+Canon requires a working installation of Docker.
+
+[Install Docker](https://docs.docker.com/engine/install/) before proceeding.
+If you are running Docker on Linux, make sure that you go through the [post installation steps](https://docs.docker.com/engine/install/linux-postinstall/).
+
+{{< tabs >}}
+{{% tab name="Linux (Ubuntu)" %}}
+
+Make sure your system has [Go 1.19](https://golangtutorial.dev/news/go-1.19-version-released/#major-changes-in-go-119-version) or later installed.
+Verify your version of Go with `go version`.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+go install github.com/viamrobotics/canon@latest
+```
+
+{{% /tab %}}
+{{% tab name="macOS" %}}
```

Comment:
I don't know if there's a step missing here or in the canon README.
When I followed these steps I got the "no profile found that contains...." error. https://viaminc.slack.com/archives/C04TJK43WA0/p1681731623189309. Not sure what steps we need to document here for the general public. @Otterverse thoughts?

Reply:
Nevermind...I think I was just not in the right directory. Leaving this comment for confirmation...

Reply:
Yeah, you'll need to be in the project folder that's generated (later on) with the cargo generate step, that pulls down the template repo https://github.com/viamrobotics/micro-rdk-robot-template

Canon wants to see a `.canon.yaml` at or above your current working directory, and uses that to determine the "root" of your project, and thus what image(s) to run and what folders to map, and that file comes from the template repo.